api/app_analytics/models.py

```diff
@@ -0,0 +1,93 @@
+from datetime import date
+
+from django.db import models
+from django.utils import timezone
+
+
+class Resource(models.IntegerChoices):
+    FLAGS = 1
+    IDENTITIES = 2
+    TRAITS = 3
+    ENVIRONMENT_DOCUMENT = 4
+
+
+class APIUsageRaw(models.Model):
+    environment_id = models.PositiveIntegerField()
+    created_at = models.DateTimeField(auto_now_add=True)
+    host = models.CharField(max_length=255)
+    resource = models.IntegerField(choices=Resource.choices)
+
+    class Meta:
+        index_together = (("environment_id", "created_at"),)
+
+
+class APIUsageAggBucketWindow(models.IntegerChoices):
+    HOUR = 1
+    DAY = 2
+
+
+class APIUsageBucket(models.Model):
+    environment_id = models.PositiveIntegerField()
+    resource = models.IntegerField(choices=Resource.choices)
+    total_count = models.PositiveIntegerField()
+    created_at = models.DateTimeField(auto_now_add=True)
+    bucket_size = models.PositiveIntegerField(help_text="Bucket size in minutes")
+
+
+# TODO: move to tasks
+def populate_bucket(bucket_size: int):
+    now = timezone.now()
+    process_from = now - timezone.timedelta(minutes=bucket_size + 1)
```

Comment:
I think this would need to be: 

```python
process_from = (now - timezone.timedelta(minutes=bucket_size + 1)).replace(second=0, microsecond=0)
```

Maybe we should also use the scheduled time of the task rather than `now` as well? But obviously that's something that will need the refactor to use a task.

Reply:
I'd say we should probably store the bucket sizes in seconds too, just in case we need the flexibility in the future (although I very much doubt it)

Reply:
Hmm... actually, I think this is slightly wrong still. To replicate what @dabeeeenster shared, we'd need something like:

```python
current_minute = timezone.now().replace(second=0, microsecond=0)
process_from = current_minute - timezone.timedelta(minutes=31 * bucket_size)
```

Note that this is purely replicating the 30m job that Ben shared - we'd need some generic way of defining how many buckets each task should process (which is obviously also a product of how often we run the task).